### ISSUE TITLE
Create preliminary videos and PDFs in DHIS2 and pull them in through our app

### ISSUE THREAD

**Issue by Ezmeyers** [iupui-soic/mHBS_tracker#39]
Tasks will include creation of adapter to handle data on the learner tracker end and modification of list activity to support handling the different data types.

**Comment by Ezmeyers** [comment]
Pdf and video uploaded for testing in resources on DHIS2. We decided to use the SDK afterall and @jmfranci has bundled it and made a broadcast with capture tracker. Once we get the commit up I can start pulling in the data.

**Comment by sunbiz** [comment]
I would advice against making changes to the SDK. We might be upgrading to a newer release of the SDK and so it might not be a good idea to make changes to the SDK. Better to make these changes in the tracker capture app or the training app.

**Comment by Ezmeyers** [comment]
Yeah, we don't want to touch the SDK, I just meant compiling it within our training app, which we don't currently have. Sorry for the lack of clarity!

**Comment by shbucher** [comment]
@Ezmeyers @sunbiz @sivaadde , I would request, as we continue to work on this issue, building upon what the MURI team did Summer 2017, that we deliberately utilize Global Health Media Project videos for demonstrative and beta-testing purposes.  

All videos are found here: https://globalhealthmedia.org/videos/ 

To start with, perhaps we can pull into mHBS/DHIS2 the following GHMP videos (each found at the link above):

--Preparing the birth room
--Preparing for the Birth of a Small Baby
--Wrap designs for skin-to-skin care

**Comment by Ezmeyers** [comment]
@sivaadde I see there is now DhisController.getInstance().getDhisApi().getResources() so I think we can pull them in that way!

**Comment by sunbiz** [comment]
We have found that you can call the Web API and access the resources. We have the two resources, one is a PDF and one is WebM video and can be accessed through https://mhbs.info/api/documents

We can also get more details like the content-type of each resource using - https://mhbs.info/api/documents/Eo5RJRxit6v

**Comment by Ezmeyers** [comment]
@sunbiz For some reason when I go to https://mhbs.info/api/documents and I am not logged in, I cannot view the page. I thought since it is public, it should be viewable...I created the XML parser and pulled in the page, but it's directing me to a blank login page at https://mhbs.info instead and pulling HTML from there. So instead of the document XML I am getting <html> <head> <body> tags from the login page.

**Comment by Ezmeyers** [comment]
Since I wasn't able to access the DHIS XML via the URL, I worked on the functionality with some other sample XML. I Created the xml parser and am able to pull/parse web xml data. I also added functionality for the videos and resources to display different views when navigating through the respective portals. I am now working on putting sample data in the lists, at least as a proof of concept until the proper XML can be pulled.

**Comment by Ezmeyers** [comment]
I am wondering if the auth isn't working because I need to use HttpsURLConnection over HttpURLConnection since the site is over HTTPS?, perhaps it wont accept the authenticator in conjunction with HTTP. This may be the incorrect route, but it's what I am currently trying since I do see there is HTTPS connection class.

**Comment by sunbiz** [comment]
@Ezmeyers, You can use HttpURLConnection to work with https from a trusted CA. We are using letsencrypt and thats a trusted CA for most reasons. Is this committed some place for me to test out and help?

**Comment by Ezmeyers** [comment]
Update; I have solved this issue, I can get the correct XML now. I will finish up this file, push the changes to my fork, and then begin populating the list! The issue was that we I couldn't use the default authentication, we had to explicitly call authentication.

**Comment by Ezmeyers** [comment]
on [my Fork](https://github.com/Ezmeyers/Training-App-Login/tree/xmlPullParser) I correctly pulled in the content type and associated with IDs. I am now going to get other textual information for items that need to populate based on resource type

**Comment by Ezmeyers** [comment]
What format did we decide to go with for Pdfs/videos? Do we have a list of them already formatted? I wanted to upload some more to resources for testing purposes. I can use dummy data, I was just wondering.

**Comment by sunbiz** [comment]
@Ezmeyers, We have two of them - one pdf and another video uploaded to mhbs.info. Please use those!

**Comment by Ezmeyers** [comment]
@sunbiz I am using those, I just wanted more, for test purposes, but I can just duplicate them.

**Comment by Ezmeyers** [comment]
@sunbiz  It looks like the Video View can't play webm format videos. Any suggestions?

**Comment by sunbiz** [comment]
VP9 coded webm is surely supported from Android 4.4 onwards. Can you check the MediaCodecInfo and check the capabilities? Any error logs that you can post when you try start().

**Comment by Ezmeyers** [comment]
All I get is:
 E/MediaPlayer: error (1, -2147483648)

and the app displays "Can't play this Video" I also tried a static webm from online and the same thing happens.
![selection_004](https://user-images.githubusercontent.com/11591786/36918899-586cf5fa-1e29-11e8-9ea6-1c35f232b5f0.png)

**Comment by Ezmeyers** [comment]
So it looks like this is the problem detailed [here](https://stackoverflow.com/questions/29776181/android-videoview-not-playing-video)

> Your url returned html code, and VideoView needed streaming resource (full path to the video).

Found out the files which were supposed to be video were grabbing login HTML which means the auth  didn't work for videos, which is weird because it works fine for PDFs so I am working on that now and will update with the fix.

**Comment by Ezmeyers** [comment]
@sunbiz  found that videos that are uploaded to DHIS2 are encoded with VP8 that is not available in the 5.1 lollipop emulator
